R/create_design.R

```diff
@@ -12,10 +12,22 @@
 #' used as reference, and `time` will be relative to the specified `at` anchor.
 #' @param when character vector of same length as `time` (or single
 #' value) determining how to interpret the provided sampling `time`. If `NULL`
-#' will use the dose time as offset (default). Other options are `cmax` or
+#' will use the dose time as offset (default). Other options are: `cmax` or
 #' `peak`, which will use the end of infusion as the base for the `time`, or
-#' `cmin` or `trough`, which will use the time of next dose as the offset.
-#' @param offset offset from standardized PK moments specified in `when`.
+#' `cmin` or `trough`, which will use the time of next dose as the offset, or
+#' `middle` or `cmid` which will use the middle between the anchored dose and
+#' the next, or `random` which takes a random time point between the anchored
+#' dose and the next.
+#' @param offset offset from standardized PK moments specified in `when`, e.g.
+#' `c(1, -1)` with `when = c("peak", "trough")` to sample 1 hour after peak and
+#' 1 hour before trough.
+#' @param scatter optional random variation in time, specified as the standard
+#' deviation, e.g. `scatter = 0.1` to allow for variation in sampling time with
+#' an SD of 0.1 hours. Only relevant for sampling times, not for
+#' regimen_update designs or target designs. Random variation does not protect
+#' for peaks or troughs becoming sampled during infusion or in previous / next
+#' dose. So value for `scatter` should be chosen appropriately and probably
+#' used in conjunction with approriate `offset` values.
```

Comment:
```suggestion
#' @param scatter optional random variation in time, specified as the standard
#' deviation, e.g. `scatter = 0.1` to allow for variation in sampling time with
#' an SD of 0.1 hours. Only implemented for sampling times (not for
#' regimen_update designs), not applicable for or target designs. Random variation 
#' does not protect for peaks or troughs becoming sampled during infusion or in 
#' previous / next dose. So value for `scatter` should be chosen appropriately and
#' probably used in conjunction with appropriate `offset` values.
```

theoretically we could use this scatter argument to account for non-exact dosing too.